File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task147_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.Arrays;
import java.util.Scanner;

public class KthLargestElement {
    public static int findKthLargest(int[] nums, int k) {
        Arrays.sort(nums);
        return nums[nums.length - k];
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] nums = new int[n];
        System.out.println("Enter the array elements:");
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }
        System.out.print("Enter the value of k: ");
        int k = scanner.nextInt();
        int result = findKthLargest(nums, k);
        System.out.println("The " + k + "th largest element is: " + result);
        scanner.close();
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task147_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return (*(int*)b - *(int*)a);
}

int findKthLargest(int* nums, int numsSize, int k) {
    qsort(nums, numsSize, sizeof(int), compare);
    return nums[k-1];
}

int main() {
    int numsSize;
    printf("Enter the number of elements: ");
    scanf("%d", &numsSize);
    
    int* nums = (int*)malloc(numsSize * sizeof(int));
    printf("Enter the array elements:\\n");
    for (int i = 0; i < numsSize; i++) {
        scanf("%d", &nums[i]);
    }
    
    int k;
    printf("Enter the value of k: ");
    scanf("%d", &k);
    
    printf("The %dth largest element is: %d\\n", k, findKthLargest(nums, numsSize, k));
    
    free(nums);
    return 0;
}
